operator/v1/types_console.go

```diff
@@ -202,6 +205,49 @@ type AddPage struct {
 	DisabledActions []string `json:"disabledActions,omitempty"`
 }
 
+type PerspectiveState string
+
+// These are valid perspective states. "Enabled" means the perspective is shown.
+// "Disabled" means the Perspective is hidden.
+// "AccessReview" means access review check is required to show or hide a Perspective.
+const (
+	Enabled      PerspectiveState = "Enabled"
+	Disabled     PerspectiveState = "Disabled"
+	AccessReview PerspectiveState = "AccessReview"
+)
+
+type PerspectiveAccessReview struct {
```

Comment:
What would happen if I don't set any of the fields within this struct? Should we enforce that at least one is set? 
```
// +kubebuilder:validation:MinItems:=1
```
(I believe that's the correct incantation)

Reply:
If we do not set any fields then we enable the perspective

Reply:
You can not specify miniItems: 1 here because it should be okay to specific only missing resources.

Reply:
Sorry, I meant `MinProperties` on the `PerspectiveAccessReview` struct would require that either Required or Missing is set, it prevents it from being an empty struct, eg https://github.com/openshift/api/blob/48d78630feb3a1df10bfd2e3797a2b96441727e2/machine/v1/types_controlplanemachineset.go#L214

Reply:
Added this validation